lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -858,8 +852,7 @@ public StatusEvent updateObjectEvent(final StatusEvent ev)
             final StatusEvent updatedEvent = updateEventForDeletion(ev);
             // if event was updated (event type gets changed to
             // StatusEventType.DELETE_ALL_VERSIONS
-
-            if (!updatedEvent.equals(ev)) {
+            if (!updatedEvent.equals(ev) || updatedEvent.getEventType().equals(StatusEventType.DELETE_ALL_VERSIONS)) {
```

Comment:
I'm sorry, but I've forgotten the driver for this change again. What was the issue this solves?

Also, I think there's another bug here - if a deletion event comes in but the object has since been undeleted, the deletion event should be updated to an undelete, right?

Reply:
what would an undelete event be?

Reply:
https://github.com/kbase/KBaseSearchEngine/blob/83b7f797f3ed221c725e7b9b384a2c2eeee83632/lib/src/kbasesearchengine/events/StatusEventType.java#L16

Reply:
do you want me to make the undelete changes to this pr though? also the line could just be:

if (updatedEvent.getEventType().equals(StatusEventType.DELETE_ALL_VERSIONS)

so that if the event is deleted, it doesnt do all the shenanigans to add to the event (which also makes workspace calls)

Reply:
No, just add the bug to the list.

I think updateEventForDeletion will *only* update to a delete event, right? If so, it might be more clear only return an event if it's update and return `null` (or `Optional.absent()`) otherwise. Then the check is `if (!updatedEvent == null | isDeletedEvent()) {` which is more readable.

I don't really have a super strong preference though.

Reply:
This ? is still unanswered I think:

> I'm sorry, but I've forgotten the driver for this change again. What was the issue this solves?

Reply:
ops fixed